src/Akka.Hosting/IAkkaHealthCheck.cs

```diff
@@ -65,7 +65,42 @@ public AkkaHealthCheckRegistration(string name, IAkkaHealthCheck instance, Healt
         _name = name;
         FailureStatus = failureStatus ?? HealthStatus.Unhealthy;
         Tags = new HashSet<string>(tags ?? [], StringComparer.OrdinalIgnoreCase);
-        _healthCheck  = instance;
+        _healthCheck  = _ => instance;
+        Timeout = timeout ?? System.Threading.Timeout.InfiniteTimeSpan;
+    }
+
+    /// <summary>
+    /// Creates a new <see cref="AkkaHealthCheckRegistration"/> template for use with DI-resolved health checks.
+    /// This constructor is intended for use with generic health check registration methods.
+    /// </summary>
+    /// <param name="name">The healthcheck name.</param>
+    /// <param name="factory">The DI-enabled factory.</param>
+    /// <param name="failureStatus">
+    /// The <see cref="HealthStatus"/> that should be reported upon failure of the health check. If the provided value
+    /// is <c>null</c>, then <see cref="HealthStatus.Unhealthy"/> will be reported.
+    /// </param>
+    /// <param name="tags">A list of tags that can be used for filtering health checks.</param>
+    /// <param name="timeout">An optional <see cref="TimeSpan"/> representing the timeout of the check.</param>
+    /// <exception cref="ArgumentNullException">Thrown if <see cref="name"/> is null.</exception>
+    /// <exception cref="ArgumentOutOfRangeException">Thrown if a negative timeout other than <see cref="System.Threading.Timeout.InfiniteTimeSpan"/> is used.</exception>
+    internal AkkaHealthCheckRegistration(string name, Func<IServiceProvider, IAkkaHealthCheck> factory, HealthStatus? failureStatus,
```

Comment:
CTOR overload which takes an `Func<IServiceProvider, IAkkaHealthCheck>` delegate . I don't see much value in making it `public` but I could be persuaded either way - the extension methods on the `AkkaConfigurationBuilder` use this for DI'd health check registrations.